### DISCUSSION TITLE
Create PDF file for download in browser

### DISCUSSION THREAD

**Discussion by olyjosh** [diegomura/react-pdf#2177] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Needed to generate PDF file and save it within the browser(Server not needed and not needed to be rendered for viewing)

**Describe the solution you'd like**
If it's possible to have something like `ReactPDF.render` work in the browser or a similar function as the component or information needed to generate the PDF does not need to make another network call to the backend.

**Comment by jeetiss** [comment, marked as answer]
use one of those components:
- `PDFViewer`
- `BlobProvider` 
- `PDFDownloadLink`

al of them exported from `@react-pdf/renderer` package

**Reply by olyjosh** [reply]
Thanks for the insight. This is helpful
